Model/Client/AbstractClient.php

```diff
@@ -199,6 +220,7 @@ protected function createRequest(Client $client): ResponseInterface
 
         if ($method === 'POST') {
             $this->params['request_id'] = $this->identityService->generateId();
+            $this->params['referrer_data'] = $this->getReferrerData();
```

Comment:
Should this only be passed on POST requests? I think we'd like to see where any request comes from

Reply:
My apologies, this was a misinterpretation on my part from the ticket description - Snow did mention all requests in the task.

I tried to make this adjustment, but it seems like your servers are perceiving this as an incorrect request if I add the JSON body to the GET requests, even though it seems like it should work fine, seeing as the request content type is still application/json.

Request:
`GET /api/v1/pa/config/payment_method_types?transaction_mode=oneoff&transaction_currency=EUR&active=1
HTTP/1.1
Content-Length: 54
User-Agent: GuzzleHttp/6.5.5 curl/7.68.0 PHP/7.3.33-1+ubuntu20.04.1+deb.sury.org+1
Host: pci-api-demo.airwallex.com
Content-Type: application/json
region: string
{"referrer_data":{"type":"magento","version":"0.1.1"}}`

Response (I removed HTML code to avoid weird formatting in GitHub, but it responded with HTML):
`HTTP/1.0 400 Bad Request
Content-Type: text/html; charset=UTF-8
Referrer-Policy: no-referrer
Content-Length: 273
Date: Thu, 05 May 2022 07:29:20 GMT
400 Bad Request, Error: Bad Request, Your client has issued a malformed or illegal request.`

For reference, a POST request, which returns the expected response correctly:
`POST /api/v1/authentication/login
HTTP/1.1
Content-Length: 106
User-Agent: GuzzleHttp/6.5.5 curl/7.68.0 PHP/7.3.33-1+ubuntu20.04.1+deb.sury.org+1
Host: pci-api-demo.airwallex.com
Content-Type: application/json
region: string
x-api-version: 2020-04-30
{"request_id":"24ddcdcb-ef01-43f7-983f-0ec839d6545f","referrer_data":{"type":"magento","version":"0.1.1"}}`

Can you please let me know if there is an issue with your servers handling GET requests with a body, or if there is some issue on our end here before we proceed further?

Reply:
Hi @laurisdedumets 

Just confirmed with our team. We should pass the referrer_data to the following calls specifically. These are all POST requests, so no need to pass it in GET requests. Apologies for the confusion caused. Please let us know if you run into other issues. 

```
payment_intents/create
pament_methods/create
customers/create
payment_consent/create
refunds/create
```

Cheers
Snow